### ISSUE TITLE
save+continue gives undefined ip index xmlinterface.class.php:260

### ISSUE THREAD

**Issue by samam9911** [chan-sccp/sccp_manager#86]
When I change anything in the sccp phone manager and click save+continue I get the error:
undefined index: ip 
File: /var/www/html/admin/modules/sccp_manager/sccpManClasses/xmlinterface.class.php:260
when I go to line 260 in the file I see this:
$xnode->$nodeName = $netValue['ip'];
![xmlinterface ip index error](https://user-images.githubusercontent.com/97066890/188334090-8279ab0c-f9c2-430f-92ba-9214ec4df164.png)
 and here is my sccp info:
![sccp info](https://user-images.githubusercontent.com/97066890/188334164-491f691f-47b8-4999-b28b-b39a1115fc5f.png)

Thanks a lot for your help and making this possible!

**Comment by samam9911** [comment]
Exception thrown with message "Fields are empty::"

Stacktrace:
#5 Exception in /var/www/html/admin/libraries/utility.functions.php:123
#4 die_freepbx in /var/www/html/admin/modules/core/Core.class.php:1432
#3 FreePBX\modules\Core:convertRequest2Array in /var/www/html/admin/modules/core/functions.inc.php:3954
#2 core_devices_add in /var/www/html/admin/modules/core/functions.inc.php:5241
#1 core_devices_configprocess in /var/www/html/admin/libraries/components.class.php:469
#0 component:processconfigpage in /var/www/html/admin/config.php:450

**Comment by MadFlasher** [comment]
Did you try to reinstall sccp_manager module ?
Look like I've had same error month ago after an upgrade of FreePBX

**Comment by samam9911** [comment]
I didn't try the reinstall. I felt it was something got corrupted in the database.
if I uninstall then install sccp_manager will I have to recreate all sccp extensions and configuration? will the uninstall remove what was configured

**Comment by numpad0** [comment]
I noticed every interface access causes 3-5 seconds of disk *writes*, and seems to throw duplicate primary key errors when interrupted. There is "BackUp Config" button in Sccp Connectivity -> Server Config -> SCCP info page, and downloaded backup file do seem to contain database files.

**Comment by samam9911** [comment]
Yes seems there is something wrong with database. Even as I delete extensions and apply config, they will still show up and I refresh my browser. Add a new sccp extension throws the same error about a field being empty then when I go back it asks me to apply config and the weirdest thing happens! the extension is showing in SCCP and as a VIRTUAL extension! I can assign a phone to it but it can NOT receive calls, if VM is enabled it goes directly to the voicemail

**Comment by samam9911** [comment]
![image](https://user-images.githubusercontent.com/97066890/194646393-86db3312-0b7a-4869-90f7-0b46d2a0b4aa.png)

**Comment by Rexz34** [comment]
I am now at this exact same problem on a fresh install. Did anyone find a way around it?

**Comment by samam9911** [comment]
I found out that uninstall the sccp-manager then install it fixes the issue but I lost all sccp-phones that were configured (of course including buttons configuration that some phones had). Wish there was a way to save all that info and restore it before doing the uninstall.

**Comment by numpad0** [comment]
https://github.com/chan-sccp/sccp_manager/issues/83#issuecomment-1230777997
``insert into sccpsettings values (`<index>`, 0, 0, `<config value>`, `<default value>`);`` worked(?) for me. Replace each <> with respective values. Looks like there can be missing rows in the database `asterisk` table `sccpmanager` (or other sccp* tables), and sccp-manager can't add them back.
